llama.cpp

```diff
@@ -13593,24 +13598,63 @@ static void llama_model_quantize_internal(const std::string & fname_inp, const s
 
     // populate the original tensors so we get an initial meta data
     for (int i = 0; i < ml.n_tensors; ++i) {
-        const struct ggml_tensor * meta = ml.get_tensor_meta(i);
-        gguf_add_tensor(ctx_out, meta);
-    }
-
-    std::ofstream fout(fname_out, std::ios::binary);
-    fout.exceptions(std::ofstream::failbit); // fail fast on write errors
-
-    const size_t meta_size = gguf_get_meta_size(ctx_out);
-
-    LLAMA_LOG_INFO("%s: meta size = %zu bytes\n", __func__, meta_size);
+        auto weight = ml.get_weight(i);
+        struct ggml_tensor * tensor = weight->tensor;
+        if (weight->idx != (ctx_outs.size() - 1) && params->keep_split) {
```

Comment:
I feel it not safe for future evolution as it assumes writing the tensors the same order they have been read. Could we simply check if `weight->idx` is not present in `ctx_outs` and retrieve `ctx_out` by tensor ?